In [2]:
import pandas as pd
df = pd.read_csv("C:\\Users\\Kudos\\Desktop\\Datascience\\Airline_Delay_Cause.csv")

In [3]:
df

year  month carrier          carrier_name airport  \
0       2022      5      9E     Endeavor Air Inc.     ABE   
1       2022      5      9E     Endeavor Air Inc.     ABY   
2       2022      5      9E     Endeavor Air Inc.     ACK   
3       2022      5      9E     Endeavor Air Inc.     AEX   
4       2022      5      9E     Endeavor Air Inc.     AGS   
...      ...    ...     ...                   ...     ...   
318012  2003      6      DL  Delta Air Lines Inc.     SEA   
318013  2003      6      DL  Delta Air Lines Inc.     SFO   
318014  2003      6      DL  Delta Air Lines Inc.     SJC   
318015  2003      6      DL  Delta Air Lines Inc.     SJU   
318016  2003      6      DL  Delta Air Lines Inc.     SLC   

                                             airport_name  arr_flights  \
0       Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                  Albany, GA: Southwest Georgia Regional         91.0   
2                       Nantucket, MA: Nantucket Memorial         19.0   
3                Alexandria, LA: Alexandria International         88.0   
4             Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                   ...          ...   
318012          Seattle, WA: Seattle/Tacoma International        480.0   
318013     San Francisco, CA: San Francisco International        505.0   
318014  San Jose, CA: Norman Y. Mineta San Jose Intern...        146.0   
318015       San Juan, PR: Luis Munoz Marin International         95.0   
318016   Salt Lake City, UT: Salt Lake City International       3466.0   

        arr_del15  carrier_ct  weather_ct  ...  security_ct  late_aircraft_ct  \
0             7.0        5.95        0.00  ...          0.0              1.00   
1            16.0        7.38        0.00  ...          0.0              6.09   
2             2.0        0.13        0.00  ...          0.0              0.88   
3            14.0        7.26        0.76  ...          0.0              1.64   
4            19.0       13.84        0.00  ...          0.0              2.09   
...           ...         ...         ...  ...          ...               ...   
318012       84.0       25.69        3.09  ...          0.0             21.26   
318013      111.0       21.78        2.24  ...          0.0             13.17   
318014       36.0        6.99        0.00  ...          0.0              2.15   
318015       13.0        3.66        0.00  ...          0.0              1.42   
318016      454.0      121.15        7.61  ...          1.0            102.33   

        arr_cancelled  arr_diverted  arr_delay  carrier_delay  weather_delay  \
0                 0.0           0.0      255.0          222.0            0.0   
1                 0.0           0.0      884.0          351.0            0.0   
2                 1.0           0.0      138.0            4.0            0.0   
3                 0.0           0.0      947.0          585.0           35.0   
4                 0.0           0.0      808.0          662.0            0.0   
...               ...           ...        ...            ...            ...   
318012            0.0           0.0     3744.0          978.0          164.0   
318013            3.0           0.0     4284.0         1376.0          138.0   
318014            0.0           0.0      896.0          205.0            0.0   
318015            0.0           0.0      367.0          120.0            0.0   
318016            9.0           0.0    16853.0         5384.0          306.0   

        nas_delay  security_delay  late_aircraft_delay  
0             4.0             0.0                 29.0  
1            81.0             0.0                452.0  
2           106.0             0.0                 28.0  
3           125.0             0.0                202.0  
4            87.0             0.0                 59.0  
...           ...             ...                  ...  
318012     1023.0             0.0               1579.

In [4]:
df.dtypes

year                     int64
month                    int64
carrier                 object
carrier_name            object
airport                 object
airport_name            object
arr_flights            float64
arr_del15              float64
carrier_ct             float64
weather_ct             float64
nas_ct                 float64
security_ct            float64
late_aircraft_ct       float64
arr_cancelled          float64
arr_diverted           float64
arr_delay              float64
carrier_delay          float64
weather_delay          float64
nas_delay              float64
security_delay         float64
late_aircraft_delay    float64
dtype: object

# Number of flights in 2022

In [5]:
flights_of_2022 = df.loc[df['year'] == 2022]

In [6]:
flights_of_2022

year  month carrier       carrier_name airport  \
0     2022      5      9E  Endeavor Air Inc.     ABE   
1     2022      5      9E  Endeavor Air Inc.     ABY   
2     2022      5      9E  Endeavor Air Inc.     ACK   
3     2022      5      9E  Endeavor Air Inc.     AEX   
4     2022      5      9E  Endeavor Air Inc.     AGS   
...    ...    ...     ...                ...     ...   
8485  2022      1      YX   Republic Airline     SYR   
8486  2022      1      YX   Republic Airline     TPA   
8487  2022      1      YX   Republic Airline     TUL   
8488  2022      1      YX   Republic Airline     VPS   
8489  2022      1      YX   Republic Airline     XNA   

                                           airport_name  arr_flights  \
0     Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                Albany, GA: Southwest Georgia Regional         91.0   
2                     Nantucket, MA: Nantucket Memorial         19.0   
3              Alexandria, LA: Alexandria International         88.0   
4           Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                 ...          ...   
8485       Syracuse, NY: Syracuse Hancock International         10.0   
8486                     Tampa, FL: Tampa International         93.0   
8487                     Tulsa, OK: Tulsa International          9.0   
8488  Valparaiso, FL: Eglin AFB Destin Fort Walton B...          3.0   
8489      Fayetteville, AR: Northwest Arkansas Regional         84.0   

      arr_del15  carrier_ct  weather_ct  ...  security_ct  late_aircraft_ct  \
0           7.0        5.95        0.00  ...         0.00              1.00   
1          16.0        7.38        0.00  ...         0.00              6.09   
2           2.0        0.13        0.00  ...         0.00              0.88   
3          14.0        7.26        0.76  ...         0.00              1.64   
4          19.0       13.84        0.00  ...         0.00              2.09   
...         ...         ...         ...  ...          ...               ...   
8485        3.0        0.92        1.17  ...         0.00              0.00   
8486       28.0        8.49        1.07  ...         0.13             12.98   
8487        3.0        1.68        0.05  ...         0.00              0.95   
8488        1.0        0.07        0.00  ...         0.00              0.93   
8489       17.0        2.75        0.13  ...         0.00              9.45   

      arr_cancelled  arr_diverted  arr_delay  carrier_delay  weather_delay  \
0               0.0           0.0      255.0          222.0            0.0   
1               0.0           0.0      884.0          351.0            0.0   
2               1.0           0.0      138.0            4.0            0.0   
3               0.0           0.0      947.0          585.0           35.0   
4               0.0           0.0      808.0          662.0            0.0   
...             ...           ...        ...            ...            ...   
8485            1.0           0.0      158.0          108.0           26.0   
8486            5.0           0.0     1740.0          492.0          120.0   
8487            0.0           0.0      195.0           33.0            8.0   
8488            0.0           0.0      123.0            9.0            0.0   
8489            6.0           0.0      956.0          131.0           11.0   

      nas_delay  security_delay  late_aircraft_delay  
0           4.0             0.0                 29.0  
1          81.0             0.0                452.0  
2         106.0             0.0                 28.0  
3         125.0             0.0                202.0  
4          87.0             0.0                 59.0  
...         ...             ...                  ...  
8485       24.0             0.0                  0.0  
8486      306.0            12.0                810.0  
8487        6.0             0.0                148.0  
8488        0.0             0.0 

In [7]:
len(flights_of_2022)

8490

# Average flight delays in 2022

In [8]:
avg_flight_delays = flights_of_2022.copy()
avg_flight_delays = avg_flight_delays.drop(avg_flight_delays.columns[[7, 8, 9, 10, 11, 12, 13, 14, 16, 17, 18, 19, 20]], axis=1)
avg_flight_delays

year  month carrier       carrier_name airport  \
0     2022      5      9E  Endeavor Air Inc.     ABE   
1     2022      5      9E  Endeavor Air Inc.     ABY   
2     2022      5      9E  Endeavor Air Inc.     ACK   
3     2022      5      9E  Endeavor Air Inc.     AEX   
4     2022      5      9E  Endeavor Air Inc.     AGS   
...    ...    ...     ...                ...     ...   
8485  2022      1      YX   Republic Airline     SYR   
8486  2022      1      YX   Republic Airline     TPA   
8487  2022      1      YX   Republic Airline     TUL   
8488  2022      1      YX   Republic Airline     VPS   
8489  2022      1      YX   Republic Airline     XNA   

                                           airport_name  arr_flights  \
0     Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                Albany, GA: Southwest Georgia Regional         91.0   
2                     Nantucket, MA: Nantucket Memorial         19.0   
3              Alexandria, LA: Alexandria International         88.0   
4           Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                 ...          ...   
8485       Syracuse, NY: Syracuse Hancock International         10.0   
8486                     Tampa, FL: Tampa International         93.0   
8487                     Tulsa, OK: Tulsa International          9.0   
8488  Valparaiso, FL: Eglin AFB Destin Fort Walton B...          3.0   
8489      Fayetteville, AR: Northwest Arkansas Regional         84.0   

      arr_delay  
0         255.0  
1         884.0  
2         138.0  
3         947.0  
4         808.0  
...         ...  
8485      158.0  
8486     1740.0  
8487      195.0  
8488      123.0  
8489      956.0  

[8490 rows x 8 columns]

In [9]:
avg_flight_delays['average_flight_delay'] = avg_flight_delays.apply(lambda row: row['arr_delay'] / row['arr_flights'] if row['arr_delay'] != 0 else float('NaN'), axis=1)
avg_flight_delays #This gives us the Number of minutes every flight has delayed on average.

year  month carrier       carrier_name airport  \
0     2022      5      9E  Endeavor Air Inc.     ABE   
1     2022      5      9E  Endeavor Air Inc.     ABY   
2     2022      5      9E  Endeavor Air Inc.     ACK   
3     2022      5      9E  Endeavor Air Inc.     AEX   
4     2022      5      9E  Endeavor Air Inc.     AGS   
...    ...    ...     ...                ...     ...   
8485  2022      1      YX   Republic Airline     SYR   
8486  2022      1      YX   Republic Airline     TPA   
8487  2022      1      YX   Republic Airline     TUL   
8488  2022      1      YX   Republic Airline     VPS   
8489  2022      1      YX   Republic Airline     XNA   

                                           airport_name  arr_flights  \
0     Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                Albany, GA: Southwest Georgia Regional         91.0   
2                     Nantucket, MA: Nantucket Memorial         19.0   
3              Alexandria, LA: Alexandria International         88.0   
4           Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                 ...          ...   
8485       Syracuse, NY: Syracuse Hancock International         10.0   
8486                     Tampa, FL: Tampa International         93.0   
8487                     Tulsa, OK: Tulsa International          9.0   
8488  Valparaiso, FL: Eglin AFB Destin Fort Walton B...          3.0   
8489      Fayetteville, AR: Northwest Arkansas Regional         84.0   

      arr_delay  average_flight_delay  
0         255.0              1.875000  
1         884.0              9.714286  
2         138.0              7.263158  
3         947.0             10.761364  
4         808.0              4.464088  
...         ...                   ...  
8485      158.0             15.800000  
8486     1740.0             18.709677  
8487      195.0             21.666667  
8488      123.0             41.000000  
8489      956.0             11.380952  

[8490 rows x 9 columns]

# Average flight delays from 2022 to 2003

In [10]:
avg_flight_delays_in_history = df.copy()
avg_flight_delays_in_history = avg_flight_delays_in_history.drop(avg_flight_delays_in_history.columns[[7, 8, 9, 10, 11, 12, 13, 14, 16, 17, 18, 19, 20]], axis=1)
avg_flight_delays_in_history

year  month carrier          carrier_name airport  \
0       2022      5      9E     Endeavor Air Inc.     ABE   
1       2022      5      9E     Endeavor Air Inc.     ABY   
2       2022      5      9E     Endeavor Air Inc.     ACK   
3       2022      5      9E     Endeavor Air Inc.     AEX   
4       2022      5      9E     Endeavor Air Inc.     AGS   
...      ...    ...     ...                   ...     ...   
318012  2003      6      DL  Delta Air Lines Inc.     SEA   
318013  2003      6      DL  Delta Air Lines Inc.     SFO   
318014  2003      6      DL  Delta Air Lines Inc.     SJC   
318015  2003      6      DL  Delta Air Lines Inc.     SJU   
318016  2003      6      DL  Delta Air Lines Inc.     SLC   

                                             airport_name  arr_flights  \
0       Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                  Albany, GA: Southwest Georgia Regional         91.0   
2                       Nantucket, MA: Nantucket Memorial         19.0   
3                Alexandria, LA: Alexandria International         88.0   
4             Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                   ...          ...   
318012          Seattle, WA: Seattle/Tacoma International        480.0   
318013     San Francisco, CA: San Francisco International        505.0   
318014  San Jose, CA: Norman Y. Mineta San Jose Intern...        146.0   
318015       San Juan, PR: Luis Munoz Marin International         95.0   
318016   Salt Lake City, UT: Salt Lake City International       3466.0   

        arr_delay  
0           255.0  
1           884.0  
2           138.0  
3           947.0  
4           808.0  
...           ...  
318012     3744.0  
318013     4284.0  
318014      896.0  
318015      367.0  
318016    16853.0  

[318017 rows x 8 columns]

In [11]:
avg_flight_delays_in_history['average_flight_delay'] = avg_flight_delays_in_history.apply(lambda row: row['arr_delay'] / row['arr_flights'] if row['arr_delay'] != 0 else float('NaN'), axis=1)
avg_flight_delays_in_history #Number of minutes delayed by every flight on average upto 2003.

year  month carrier          carrier_name airport  \
0       2022      5      9E     Endeavor Air Inc.     ABE   
1       2022      5      9E     Endeavor Air Inc.     ABY   
2       2022      5      9E     Endeavor Air Inc.     ACK   
3       2022      5      9E     Endeavor Air Inc.     AEX   
4       2022      5      9E     Endeavor Air Inc.     AGS   
...      ...    ...     ...                   ...     ...   
318012  2003      6      DL  Delta Air Lines Inc.     SEA   
318013  2003      6      DL  Delta Air Lines Inc.     SFO   
318014  2003      6      DL  Delta Air Lines Inc.     SJC   
318015  2003      6      DL  Delta Air Lines Inc.     SJU   
318016  2003      6      DL  Delta Air Lines Inc.     SLC   

                                             airport_name  arr_flights  \
0       Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                  Albany, GA: Southwest Georgia Regional         91.0   
2                       Nantucket, MA: Nantucket Memorial         19.0   
3                Alexandria, LA: Alexandria International         88.0   
4             Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                   ...          ...   
318012          Seattle, WA: Seattle/Tacoma International        480.0   
318013     San Francisco, CA: San Francisco International        505.0   
318014  San Jose, CA: Norman Y. Mineta San Jose Intern...        146.0   
318015       San Juan, PR: Luis Munoz Marin International         95.0   
318016   Salt Lake City, UT: Salt Lake City International       3466.0   

        arr_delay  average_flight_delay  
0           255.0              1.875000  
1           884.0              9.714286  
2           138.0              7.263158  
3           947.0             10.761364  
4           808.0              4.464088  
...           ...                   ...  
318012     3744.0              7.800000  
318013     4284.0              8.483168  
318014      896.0              6.136986  
318015      367.0              3.863158  
318016    16853.0              4.862377  

[318017 rows x 9 columns]

# List of all Unique Airlines

In [12]:
unique_airlines = df['carrier_name'].unique()
list_of_airlines = pd.DataFrame({"Unique_Airlines":unique_airlines})

In [13]:
list_of_airlines

Unique_Airlines
0              Endeavor Air Inc.
1         American Airlines Inc.
2           Alaska Airlines Inc.
3                JetBlue Airways
4           Delta Air Lines Inc.
5         Frontier Airlines Inc.
6                  Allegiant Air
7         Hawaiian Airlines Inc.
8                      Envoy Air
9               Spirit Air Lines
10             PSA Airlines Inc.
11         SkyWest Airlines Inc.
12                   Horizon Air
13         United Air Lines Inc.
14        Southwest Airlines Co.
15            Mesa Airlines Inc.
16              Republic Airline
17       ExpressJet Airlines LLC
18      ExpressJet Airlines Inc.
19                Virgin America
20               US Airways Inc.
21   AirTran Airways Corporation
22  American Eagle Airlines Inc.
23        Pinnacle Airlines Inc.
24   Atlantic Southeast Airlines
25    Continental Air Lines Inc.
26                   Comair Inc.
27       Northwest Airlines Inc.
28           Aloha Airlines Inc.
29        ATA Airlines d/b/a ATA
30              Independence Air
31    America West Airlines Inc.
32       Atlantic Coast Airlines

In [14]:
start_year = 2022
end_year = 2012
year_range = df.loc[(df['year'] >= end_year) & (df['year'] <= start_year)]
year_range

year  month carrier        carrier_name airport  \
0       2022      5      9E   Endeavor Air Inc.     ABE   
1       2022      5      9E   Endeavor Air Inc.     ABY   
2       2022      5      9E   Endeavor Air Inc.     ACK   
3       2022      5      9E   Endeavor Air Inc.     AEX   
4       2022      5      9E   Endeavor Air Inc.     AGS   
...      ...    ...     ...                 ...     ...   
168098  2012      1      YV  Mesa Airlines Inc.     SYR   
168099  2012      1      YV  Mesa Airlines Inc.     TLH   
168100  2012      1      YV  Mesa Airlines Inc.     TUS   
168101  2012      1      YV  Mesa Airlines Inc.     TYS   
168102  2012      1      YV  Mesa Airlines Inc.     YUM   

                                             airport_name  arr_flights  \
0       Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                  Albany, GA: Southwest Georgia Regional         91.0   
2                       Nantucket, MA: Nantucket Memorial         19.0   
3                Alexandria, LA: Alexandria International         88.0   
4             Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                   ...          ...   
168098       Syracuse, NY: Syracuse Hancock International         58.0   
168099         Tallahassee, FL: Tallahassee International         16.0   
168100                   Tucson, AZ: Tucson International        239.0   
168101                        Knoxville, TN: McGhee Tyson          4.0   
168102             Yuma, AZ: Yuma MCAS/Yuma International        101.0   

        arr_del15  carrier_ct  weather_ct  ...  security_ct  late_aircraft_ct  \
0             7.0        5.95        0.00  ...          0.0              1.00   
1            16.0        7.38        0.00  ...          0.0              6.09   
2             2.0        0.13        0.00  ...          0.0              0.88   
3            14.0        7.26        0.76  ...          0.0              1.64   
4            19.0       13.84        0.00  ...          0.0              2.09   
...           ...         ...         ...  ...          ...               ...   
168098        7.0        1.00        0.00  ...          0.0              4.24   
168099        4.0        2.00        0.00  ...          0.0              1.83   
168100       25.0        6.00        1.00  ...          0.0             16.21   
168101        0.0        0.00        0.00  ...          0.0              0.00   
168102       17.0        1.00        0.00  ...          0.0             13.14   

        arr_cancelled  arr_diverted  arr_delay  carrier_delay  weather_delay  \
0                 0.0           0.0      255.0          222.0            0.0   
1                 0.0           0.0      884.0          351.0            0.0   
2                 1.0           0.0      138.0            4.0            0.0   
3                 0.0           0.0      947.0          585.0           35.0   
4                 0.0           0.0      808.0          662.0            0.0   
...               ...           ...        ...            ...            ...   
168098            1.0           1.0      683.0           82.0            0.0   
168099            3.0           0.0      152.0           72.0            0.0   
168100            0.0           0.0      960.0          215.0           61.0   
168101            1.0           0.0        0.0            0.0            0.0   
168102            0.0           0.0      558.0           79.0            0.0   

        nas_delay  security_delay  late_aircraft_delay  
0             4.0             0.0                 29.0  
1            81.0             0.0                452.0  
2           106.0             0.0                 28.0  
3           125.0             0.0                202.0  
4            87.0             0.0                 59.0  
...           ...             ...                  ...  
168098      121.0             0.0                480.0  
168099        6.0   

In [15]:
filter_value = 'Endeavor Air Inc.'
EAC_in_range = year_range.loc[year_range['carrier_name'] == filter_value]
max_endeavor_arr_flights = EAC_in_range['arr_flights'].max()
max_endeavor_arr_flights #A way to get the max arr_flight value for one airlines

6629.0

In [16]:
max_arr_flights = df.groupby("carrier_name")["arr_flights"].max() #Using groupby function to implement the same thing for all airlines
max_arr_flights

carrier_name
ATA Airlines d/b/a ATA           2563.0
AirTran Airways Corporation      8178.0
Alaska Airlines Inc.             6572.0
Allegiant Air                    1184.0
Aloha Airlines Inc.              2029.0
America West Airlines Inc.       5938.0
American Airlines Inc.          15097.0
American Eagle Airlines Inc.     8882.0
Atlantic Coast Airlines          9009.0
Atlantic Southeast Airlines     14088.0
Comair Inc.                     11936.0
Continental Air Lines Inc.       8075.0
Delta Air Lines Inc.            21977.0
Endeavor Air Inc.                6629.0
Envoy Air                        7063.0
ExpressJet Airlines Inc.        10384.0
ExpressJet Airlines LLC          3696.0
Frontier Airlines Inc.           4665.0
Hawaiian Airlines Inc.           3136.0
Horizon Air                      3004.0
Independence Air                 9572.0
JetBlue Airways                  5233.0
Mesa Airlines Inc.               3888.0
Northwest Airlines Inc.         10287.0
PSA Airlines Inc.          

In [17]:
max_arr_flights.max() #To see the maximum value in the list

21977.0

In [18]:
max_arr_flights[max_arr_flights == max_arr_flights.max()].index[0] #To see the associated airline with the maximum value

'Delta Air Lines Inc.'

## Therefore, we can see that Delta Air Lines Inc. has the maximum number of delayed air flights in the last 10 years in a single month. Now let's find their total number of delayed flights in the last 10 years combined.

In [19]:
year_range

year  month carrier        carrier_name airport  \
0       2022      5      9E   Endeavor Air Inc.     ABE   
1       2022      5      9E   Endeavor Air Inc.     ABY   
2       2022      5      9E   Endeavor Air Inc.     ACK   
3       2022      5      9E   Endeavor Air Inc.     AEX   
4       2022      5      9E   Endeavor Air Inc.     AGS   
...      ...    ...     ...                 ...     ...   
168098  2012      1      YV  Mesa Airlines Inc.     SYR   
168099  2012      1      YV  Mesa Airlines Inc.     TLH   
168100  2012      1      YV  Mesa Airlines Inc.     TUS   
168101  2012      1      YV  Mesa Airlines Inc.     TYS   
168102  2012      1      YV  Mesa Airlines Inc.     YUM   

                                             airport_name  arr_flights  \
0       Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                  Albany, GA: Southwest Georgia Regional         91.0   
2                       Nantucket, MA: Nantucket Memorial         19.0   
3                Alexandria, LA: Alexandria International         88.0   
4             Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                   ...          ...   
168098       Syracuse, NY: Syracuse Hancock International         58.0   
168099         Tallahassee, FL: Tallahassee International         16.0   
168100                   Tucson, AZ: Tucson International        239.0   
168101                        Knoxville, TN: McGhee Tyson          4.0   
168102             Yuma, AZ: Yuma MCAS/Yuma International        101.0   

        arr_del15  carrier_ct  weather_ct  ...  security_ct  late_aircraft_ct  \
0             7.0        5.95        0.00  ...          0.0              1.00   
1            16.0        7.38        0.00  ...          0.0              6.09   
2             2.0        0.13        0.00  ...          0.0              0.88   
3            14.0        7.26        0.76  ...          0.0              1.64   
4            19.0       13.84        0.00  ...          0.0              2.09   
...           ...         ...         ...  ...          ...               ...   
168098        7.0        1.00        0.00  ...          0.0              4.24   
168099        4.0        2.00        0.00  ...          0.0              1.83   
168100       25.0        6.00        1.00  ...          0.0             16.21   
168101        0.0        0.00        0.00  ...          0.0              0.00   
168102       17.0        1.00        0.00  ...          0.0             13.14   

        arr_cancelled  arr_diverted  arr_delay  carrier_delay  weather_delay  \
0                 0.0           0.0      255.0          222.0            0.0   
1                 0.0           0.0      884.0          351.0            0.0   
2                 1.0           0.0      138.0            4.0            0.0   
3                 0.0           0.0      947.0          585.0           35.0   
4                 0.0           0.0      808.0          662.0            0.0   
...               ...           ...        ...            ...            ...   
168098            1.0           1.0      683.0           82.0            0.0   
168099            3.0           0.0      152.0           72.0            0.0   
168100            0.0           0.0      960.0          215.0           61.0   
168101            1.0           0.0        0.0            0.0            0.0   
168102            0.0           0.0      558.0           79.0            0.0   

        nas_delay  security_delay  late_aircraft_delay  
0             4.0             0.0                 29.0  
1            81.0             0.0                452.0  
2           106.0             0.0                 28.0  
3           125.0             0.0                202.0  
4            87.0             0.0                 59.0  
...           ...             ...                  ...  
168098      121.0             0.0                480.0  
168099        6.0   

In [20]:
delta_airlines = year_range['carrier_name'] == 'Delta Air Lines Inc.' #This filters out all the delta airlines rows from the last 10 years
table_of_all_delta_airlines = year_range[delta_airlines] #This puts them in a new data frame.
table_of_all_delta_airlines

year  month carrier          carrier_name airport  \
352     2022      5      DL  Delta Air Lines Inc.     ABQ   
353     2022      5      DL  Delta Air Lines Inc.     ALB   
354     2022      5      DL  Delta Air Lines Inc.     ANC   
355     2022      5      DL  Delta Air Lines Inc.     ATL   
356     2022      5      DL  Delta Air Lines Inc.     ATW   
...      ...    ...     ...                   ...     ...   
167194  2012      1      DL  Delta Air Lines Inc.     TPA   
167195  2012      1      DL  Delta Air Lines Inc.     TUL   
167196  2012      1      DL  Delta Air Lines Inc.     TUS   
167197  2012      1      DL  Delta Air Lines Inc.     TYS   
167198  2012      1      DL  Delta Air Lines Inc.     VPS   

                                             airport_name  arr_flights  \
352     Albuquerque, NM: Albuquerque International Sun...         93.0   
353                      Albany, NY: Albany International         93.0   
354     Anchorage, AK: Ted Stevens Anchorage Internati...        215.0   
355     Atlanta, GA: Hartsfield-Jackson Atlanta Intern...      18297.0   
356                  Appleton, WI: Appleton International         31.0   
...                                                   ...          ...   
167194                     Tampa, FL: Tampa International        865.0   
167195                     Tulsa, OK: Tulsa International         22.0   
167196                   Tucson, AZ: Tucson International         76.0   
167197                        Knoxville, TN: McGhee Tyson         28.0   
167198  Valparaiso, FL: Eglin AFB Destin Fort Walton B...         25.0   

        arr_del15  carrier_ct  weather_ct  ...  security_ct  late_aircraft_ct  \
352          12.0        6.64        1.00  ...         0.00              4.25   
353          28.0       16.07        0.00  ...         0.00              5.87   
354          25.0       17.90        0.00  ...         0.00              4.16   
355        2573.0     1012.39       95.51  ...         0.52            963.54   
356           3.0        2.00        0.00  ...         0.00              0.00   
...           ...         ...         ...  ...          ...               ...   
167194       93.0       27.38        4.55  ...         0.00             34.60   
167195        5.0        2.95        0.00  ...         0.00              1.46   
167196        6.0        1.31        0.00  ...         0.00              1.94   
167197        6.0        2.09        0.00  ...         0.00              2.93   
167198        3.0        0.93        0.00  ...         0.00              0.80   

        arr_cancelled  arr_diverted  arr_delay  carrier_delay  weather_delay  \
352               2.0           0.0      429.0          218.0           49.0   
353               5.0           0.0     1374.0          602.0            0.0   
354               2.0           1.0      958.0          676.0            0.0   
355             444.0          23.0   193784.0       102534.0         7741.0   
356               0.0           0.0      109.0           89.0            0.0   
...               ...           ...        ...            ...            ...   
167194            3.0           0.0     4233.0         1421.0          190.0   
167195            0.0           0.0      336.0          266.0            0.0   
167196            0.0           1.0      209.0           41.0            0.0   
167197            0.0           0.0      286.0           94.0            0.0   
167198            0.0           0.0      223.0          149.0            0.0   

        nas_delay  security_delay  late_aircraft_delay  
352           4.0             0.0                158.0  
353         199.0             0.0                573.0  
354          63.0             0.0                219.0  
355       19640.0            34.0              63835.0  
356          20.0             0.0                  0.0  
...           ...             ...                  ...  
167194      870.0             0.0               1752.

In [21]:
sum_delta_arr_flights = table_of_all_delta_airlines['arr_flights'].sum()
sum_delta_arr_flights

8630939.0

# Total number of delayed flights by Delta Airlines Inc. in the last 10 years is 8630939.0.

In [22]:
filter_value = 'Endeavor Air Inc.'
EAC_in_range = year_range.loc[year_range['carrier_name'] == filter_value]
min_endeavor_arr_flights = EAC_in_range['arr_flights'].min()
min_endeavor_arr_flights #A way to get the min arr_flight value for one airlines

1.0

In [23]:
min_arr_flights = df.groupby("carrier_name")["arr_flights"].min() #Using groupby function to implement the same thing for all airlines
min_arr_flights

carrier_name
ATA Airlines d/b/a ATA           1.0
AirTran Airways Corporation      1.0
Alaska Airlines Inc.             1.0
Allegiant Air                    1.0
Aloha Airlines Inc.             16.0
America West Airlines Inc.       1.0
American Airlines Inc.           1.0
American Eagle Airlines Inc.     1.0
Atlantic Coast Airlines          1.0
Atlantic Southeast Airlines      1.0
Comair Inc.                      1.0
Continental Air Lines Inc.       1.0
Delta Air Lines Inc.             1.0
Endeavor Air Inc.                1.0
Envoy Air                        1.0
ExpressJet Airlines Inc.         1.0
ExpressJet Airlines LLC          1.0
Frontier Airlines Inc.           1.0
Hawaiian Airlines Inc.           1.0
Horizon Air                      1.0
Independence Air                 1.0
JetBlue Airways                  1.0
Mesa Airlines Inc.               1.0
Northwest Airlines Inc.          1.0
PSA Airlines Inc.                1.0
Pinnacle Airlines Inc.           1.0
Republic Airline         

## Since the lowest number of flight delays are 1, we can't select here because it might be that they even have a lower number of flights in general. So we pick the highest minimum 'Aloha Airlines Inc. '

In [24]:
highest_in_minimums = min_arr_flights.max()
highest_in_minimums

16.0

In [25]:
min_arr_flights[min_arr_flights == min_arr_flights.max()].index[0] #To find the airlines that has the highest arr_flights among the minimum list

'Aloha Airlines Inc.'

In [26]:
aloha_airlines = year_range['carrier_name'] == 'Aloha Airlines Inc.' #This filters out all the aloha airlines rows from the last 10 years
table_of_all_aloha_airlines = year_range[aloha_airlines] #This puts them in a new data frame.
table_of_all_aloha_airlines

Empty DataFrame
Columns: [year, month, carrier, carrier_name, airport, airport_name, arr_flights, arr_del15, carrier_ct, weather_ct, nas_ct, security_ct, late_aircraft_ct, arr_cancelled, arr_diverted, arr_delay, carrier_delay, weather_delay, nas_delay, security_delay, late_aircraft_delay]
Index: []

[0 rows x 21 columns]

In [27]:
sum_Aloha_arr_flights = table_of_all_aloha_airlines['arr_flights'].sum()
sum_Aloha_arr_flights

0.0

### This signals that Aloha Airlines Inc. has not had any delayed flights in the last 10 years. It could be because the airlines stopped sending flights, which is not relevant to the study. So we implement it with an actively working airlines in the same minimum set of data for the last 10 years, 'AirTran Airways Corporation'.

In [28]:
AirTran_airlines = year_range['carrier_name'] == 'AirTran Airways Corporation' #This filters out all the aloha airlines rows from the last 10 years
table_of_all_AirTran = year_range[AirTran_airlines] #This puts them in a new data frame.
table_of_all_AirTran

year  month carrier                 carrier_name airport  \
124230  2014     12      FL  AirTran Airways Corporation     ATL   
124231  2014     12      FL  AirTran Airways Corporation     BWI   
124232  2014     12      FL  AirTran Airways Corporation     MDW   
124233  2014     12      FL  AirTran Airways Corporation     BOS   
124234  2014     12      FL  AirTran Airways Corporation     JAX   
...      ...    ...     ...                          ...     ...   
167460  2012      1      FL  AirTran Airways Corporation     SJU   
167461  2012      1      FL  AirTran Airways Corporation     SRQ   
167462  2012      1      FL  AirTran Airways Corporation     STL   
167463  2012      1      FL  AirTran Airways Corporation     TPA   
167464  2012      1      FL  AirTran Airways Corporation     TYS   

                                             airport_name  arr_flights  \
124230  Atlanta, GA: Hartsfield-Jackson Atlanta Intern...       1361.0   
124231  Baltimore, MD: Baltimore/Washington Internatio...         83.0   
124232          Chicago, IL: Chicago Midway International        101.0   
124233                    Boston, MA: Logan International         76.0   
124234       Jacksonville, FL: Jacksonville International         56.0   
...                                                   ...          ...   
167460       San Juan, PR: Luis Munoz Marin International        251.0   
167461  Sarasota/Bradenton, FL: Sarasota/Bradenton Int...        205.0   
167462      St. Louis, MO: St Louis Lambert International         97.0   
167463                     Tampa, FL: Tampa International        609.0   
167464                        Knoxville, TN: McGhee Tyson         20.0   

        arr_del15  carrier_ct  weather_ct  ...  security_ct  late_aircraft_ct  \
124230      167.0       29.10        0.81  ...          0.0             70.86   
124231       10.0        4.30        0.00  ...          0.0              3.66   
124232       10.0        4.55        0.00  ...          0.0              4.96   
124233        5.0        2.60        0.00  ...          0.0              1.89   
124234       13.0        7.42        0.00  ...          0.0              4.83   
...           ...         ...         ...  ...          ...               ...   
167460       16.0        6.09        0.86  ...          0.0              5.00   
167461       27.0        3.47        0.00  ...          0.0              6.47   
167462       12.0        3.00        0.00  ...          0.0              5.53   
167463       65.0        9.90        2.00  ...          0.0             14.92   
167464        3.0        1.83        0.00  ...          0.0              1.00   

        arr_cancelled  arr_diverted  arr_delay  carrier_delay  weather_delay  \
124230            9.0           0.0     7989.0         1806.0           38.0   
124231            0.0           0.0      398.0          154.0            0.0   
124232            0.0           0.0      434.0          203.0            0.0   
124233            0.0           0.0      219.0          159.0            0.0   
124234            0.0           0.0      502.0          224.0            0.0   
...               ...           ...        ...            ...            ...   
167460            0.0           1.0      708.0          202.0           56.0   
167461            1.0           0.0      876.0           81.0            0.0   
167462            0.0           0.0      371.0           61.0            0.0   
167463            1.0           0.0     3143.0          540.0          317.0   
167464            0.0           0.0       95.0           73.0            0.0   

        nas_delay  security_delay  late_aircraft_delay  
124230     2062.0             0.0               4083.0  
124231       44.0             0.0                200.0  
124232       29.0             0.0                202.0  
124233       10.0             0.0                 50.0  
124234       28.0             0.0                250.0  
...           ...         

In [29]:
sum_AirTran_arr_flights = table_of_all_AirTran['arr_flights'].sum()
sum_AirTran_arr_flights

471609.0

# One of the flights with the lowest number of arr_flights has a sum of 471609.0 flights in the last 10 years. Which is significantly lower than Delta Airlines Inc. with a value of 8630939.0.

# Now we do the same process in terms of airports where aircrafts have landed.

### List of all unique airports

In [30]:
unique_airports = df['airport_name'].unique()
list_of_airports = pd.DataFrame({"unique_airports":unique_airports}) #Puts the unique airlines in a separate dataframe
list_of_airports

unique_airports
0    Allentown/Bethlehem/Easton, PA: Lehigh Valley ...
1               Albany, GA: Southwest Georgia Regional
2                    Nantucket, MA: Nantucket Memorial
3             Alexandria, LA: Alexandria International
4          Augusta, GA: Augusta Regional at Bush Field
..                                                 ...
415                              Hoolehua, HI: Molokai
416                         Killeen, TX: Skylark Field
417                         Spokane, WA: Fairchild AFB
418                         Columbus, MS: Columbus AFB
419                     Unalaska, AK: Unalaska Airport

[420 rows x 1 columns]

In [31]:
year_range

year  month carrier        carrier_name airport  \
0       2022      5      9E   Endeavor Air Inc.     ABE   
1       2022      5      9E   Endeavor Air Inc.     ABY   
2       2022      5      9E   Endeavor Air Inc.     ACK   
3       2022      5      9E   Endeavor Air Inc.     AEX   
4       2022      5      9E   Endeavor Air Inc.     AGS   
...      ...    ...     ...                 ...     ...   
168098  2012      1      YV  Mesa Airlines Inc.     SYR   
168099  2012      1      YV  Mesa Airlines Inc.     TLH   
168100  2012      1      YV  Mesa Airlines Inc.     TUS   
168101  2012      1      YV  Mesa Airlines Inc.     TYS   
168102  2012      1      YV  Mesa Airlines Inc.     YUM   

                                             airport_name  arr_flights  \
0       Allentown/Bethlehem/Easton, PA: Lehigh Valley ...        136.0   
1                  Albany, GA: Southwest Georgia Regional         91.0   
2                       Nantucket, MA: Nantucket Memorial         19.0   
3                Alexandria, LA: Alexandria International         88.0   
4             Augusta, GA: Augusta Regional at Bush Field        181.0   
...                                                   ...          ...   
168098       Syracuse, NY: Syracuse Hancock International         58.0   
168099         Tallahassee, FL: Tallahassee International         16.0   
168100                   Tucson, AZ: Tucson International        239.0   
168101                        Knoxville, TN: McGhee Tyson          4.0   
168102             Yuma, AZ: Yuma MCAS/Yuma International        101.0   

        arr_del15  carrier_ct  weather_ct  ...  security_ct  late_aircraft_ct  \
0             7.0        5.95        0.00  ...          0.0              1.00   
1            16.0        7.38        0.00  ...          0.0              6.09   
2             2.0        0.13        0.00  ...          0.0              0.88   
3            14.0        7.26        0.76  ...          0.0              1.64   
4            19.0       13.84        0.00  ...          0.0              2.09   
...           ...         ...         ...  ...          ...               ...   
168098        7.0        1.00        0.00  ...          0.0              4.24   
168099        4.0        2.00        0.00  ...          0.0              1.83   
168100       25.0        6.00        1.00  ...          0.0             16.21   
168101        0.0        0.00        0.00  ...          0.0              0.00   
168102       17.0        1.00        0.00  ...          0.0             13.14   

        arr_cancelled  arr_diverted  arr_delay  carrier_delay  weather_delay  \
0                 0.0           0.0      255.0          222.0            0.0   
1                 0.0           0.0      884.0          351.0            0.0   
2                 1.0           0.0      138.0            4.0            0.0   
3                 0.0           0.0      947.0          585.0           35.0   
4                 0.0           0.0      808.0          662.0            0.0   
...               ...           ...        ...            ...            ...   
168098            1.0           1.0      683.0           82.0            0.0   
168099            3.0           0.0      152.0           72.0            0.0   
168100            0.0           0.0      960.0          215.0           61.0   
168101            1.0           0.0        0.0            0.0            0.0   
168102            0.0           0.0      558.0           79.0            0.0   

        nas_delay  security_delay  late_aircraft_delay  
0             4.0             0.0                 29.0  
1            81.0             0.0                452.0  
2           106.0             0.0                 28.0  
3           125.0             0.0                202.0  
4            87.0             0.0                 59.0  
...           ...             ...                  ...  
168098      121.0             0.0                480.0  
168099        6.0   

In [32]:
max_airport_arr_flights = df.groupby("airport_name")["arr_flights"].max()
max_airport_arr_flights

airport_name
Aberdeen, SD: Aberdeen Regional                       75.0
Abilene, TX: Abilene Regional                        244.0
Adak Island, AK: Adak                                  9.0
Aguadilla, PR: Rafael Hernandez                      155.0
Akron, OH: Akron-Canton Regional                     461.0
                                                     ...  
Wrangell, AK: Wrangell Airport                        62.0
Yakima, WA: Yakima Air Terminal/McAllister Field      83.0
Yakutat, AK: Yakutat Airport                          62.0
Youngstown/Warren, OH: Youngstown-Warren Regional      2.0
Yuma, AZ: Yuma MCAS/Yuma International               304.0
Name: arr_flights, Length: 420, dtype: float64

In [33]:
max_airport_arr_flights.max() # This indicates the airport data row which shows the maximum number of arr_flights within the entire data set, and its not a sum of anything.

21977.0

In [34]:
max_airport_arr_flights[max_airport_arr_flights == max_airport_arr_flights.max()].index[0] # Associated airport name with the maximum number of arr_flights

'Atlanta, GA: Hartsfield-Jackson Atlanta International'

## Therefore, we can see that Atlanta, GA: Hartsfield-Jackson Atlanta International airport has the maximum number of delayed air flights in the last 10 years in a single month. Now let's find the total number of delayed flights from this airport in the last 10 years combined.

In [35]:
HJAI_airport = year_range['airport_name'] == 'Atlanta, GA: Hartsfield-Jackson Atlanta International' #This filters out all the delta airlines rows from the last 10 years
table_of_all_HJAI_airport = year_range[HJAI_airport] #This puts them in a new data frame.
table_of_all_HJAI_airport

year  month carrier                  carrier_name airport  \
6       2022      5      9E             Endeavor Air Inc.     ATL   
102     2022      5      AA        American Airlines Inc.     ATL   
209     2022      5      AS          Alaska Airlines Inc.     ATL   
288     2022      5      B6               JetBlue Airways     ATL   
355     2022      5      DL          Delta Air Lines Inc.     ATL   
...      ...    ...     ...                           ...     ...   
167488  2012      1      MQ  American Eagle Airlines Inc.     ATL   
167626  2012      1      OO         SkyWest Airlines Inc.     ATL   
167772  2012      1      UA         United Air Lines Inc.     ATL   
167857  2012      1      US               US Airways Inc.     ATL   
168018  2012      1      YV            Mesa Airlines Inc.     ATL   

                                             airport_name  arr_flights  \
6       Atlanta, GA: Hartsfield-Jackson Atlanta Intern...       3042.0   
102     Atlanta, GA: Hartsfield-Jackson Atlanta Intern...        529.0   
209     Atlanta, GA: Hartsfield-Jackson Atlanta Intern...         84.0   
288     Atlanta, GA: Hartsfield-Jackson Atlanta Intern...        196.0   
355     Atlanta, GA: Hartsfield-Jackson Atlanta Intern...      18297.0   
...                                                   ...          ...   
167488  Atlanta, GA: Hartsfield-Jackson Atlanta Intern...        435.0   
167626  Atlanta, GA: Hartsfield-Jackson Atlanta Intern...        275.0   
167772  Atlanta, GA: Hartsfield-Jackson Atlanta Intern...        103.0   
167857  Atlanta, GA: Hartsfield-Jackson Atlanta Intern...        458.0   
168018  Atlanta, GA: Hartsfield-Jackson Atlanta Intern...        189.0   

        arr_del15  carrier_ct  weather_ct  ...  security_ct  late_aircraft_ct  \
6           453.0      142.96       13.53  ...         0.00            189.53   
102         157.0       56.03        7.15  ...         1.66             65.52   
209          13.0        4.48        0.00  ...         0.28              4.43   
288          43.0       27.67        1.40  ...         0.00             11.00   
355        2573.0     1012.39       95.51  ...         0.52            963.54   
...           ...         ...         ...  ...          ...               ...   
167488       84.0       22.15       11.82  ...         0.00             13.40   
167626       56.0       12.73        0.32  ...         0.00             21.71   
167772       17.0        6.30        0.93  ...         0.00              4.46   
167857       75.0       22.51        0.00  ...         0.80             16.61   
168018       31.0        9.00        0.00  ...         0.00              7.69   

        arr_cancelled  arr_diverted  arr_delay  carrier_delay  weather_delay  \
6                 2.0           3.0    38397.0        20775.0         1205.0   
102               6.0           0.0     9921.0         3223.0          657.0   
209               2.0           0.0      342.0          114.0            0.0   
288               3.0           0.0     3021.0         1645.0          161.0   
355             444.0          23.0   193784.0       102534.0         7741.0   
...               ...           ...        ...            ...            ...   
167488           13.0           2.0     3690.0          883.0          516.0   
167626            3.0           1.0     3374.0          627.0           93.0   
167772            1.0           0.0      679.0          198.0           69.0   
167857            3.0           0.0     3268.0         1107.0            0.0   
168018            6.0           1.0     1509.0          719.0            0.0   

        nas_delay  security_delay  late_aircraft_delay  
6          3610.0             0.0              12807.0  
102        1041.0            41.0               4959.0  
209          85.0             5.0                138.0  
288         240.0             0.0                975.0  
355       19640.0            34.0              63835.0  
...           

In [36]:
sum_HJAI_arr_flights = table_of_all_HJAI_airport['arr_flights'].sum()
sum_HJAI_arr_flights

3736583.0

# Total number of delayed flights by Atlanta, GA: Hartsfield-Jackson Atlanta International airport in the last 10 years is 3736583.0.

In [37]:
filter_value_airport = 'Atlanta, GA: Hartsfield-Jackson Atlanta International'
HJAI_in_range = year_range.loc[year_range['airport_name'] == filter_value_airport]
min_HJAI_arr_flights = EAC_in_range['arr_flights'].min()
min_HJAI_arr_flights #A way to get the min arr_flight value for one airport

1.0

In [38]:
min_arr_flights_airport = df.groupby("airport_name")["arr_flights"].min() #Using groupby function to implement the same thing for all airports
min_arr_flights_airport

airport_name
Aberdeen, SD: Aberdeen Regional                       2.0
Abilene, TX: Abilene Regional                         1.0
Adak Island, AK: Adak                                 7.0
Aguadilla, PR: Rafael Hernandez                       1.0
Akron, OH: Akron-Canton Regional                      1.0
                                                     ... 
Wrangell, AK: Wrangell Airport                       56.0
Yakima, WA: Yakima Air Terminal/McAllister Field     25.0
Yakutat, AK: Yakutat Airport                         55.0
Youngstown/Warren, OH: Youngstown-Warren Regional     2.0
Yuma, AZ: Yuma MCAS/Yuma International                1.0
Name: arr_flights, Length: 420, dtype: float64

## Since the lowest number of flight delays are 1, we can't select here because there might be a lower number of landings in general. So we pick the highest minimum 'Sanford, FL: Orlando Sanford International'.

In [39]:
highest_in_minimums_airport = min_arr_flights_airport.max()
highest_in_minimums_airport

364.0

In [40]:
min_arr_flights_airport[min_arr_flights_airport == min_arr_flights_airport.max()].index[0] # To find the airport that has the highest arr_flights among the minimum list

'Sanford, FL: Orlando Sanford International'

In [41]:
OSI_airport = year_range['airport_name'] == 'Sanford, FL: Orlando Sanford International' #This filters out all the aloha airlines rows from the last 10 years
table_of_all_OSI_airport = year_range[OSI_airport] #This puts them in a new data frame.
table_of_all_OSI_airport

year  month carrier   carrier_name airport  \
689    2022      5      G4  Allegiant Air     SFB   
2395   2022      4      G4  Allegiant Air     SFB   
4105   2022      3      G4  Allegiant Air     SFB   
5794   2022      2      G4  Allegiant Air     SFB   
7478   2022      1      G4  Allegiant Air     SFB   
9178   2021     12      G4  Allegiant Air     SFB   
10882  2021     11      G4  Allegiant Air     SFB   
12602  2021     10      G4  Allegiant Air     SFB   
14362  2021      9      G4  Allegiant Air     SFB   
16102  2021      8      G4  Allegiant Air     SFB   
17820  2021      7      G4  Allegiant Air     SFB   
19525  2021      6      G4  Allegiant Air     SFB   
21214  2021      5      G4  Allegiant Air     SFB   
22829  2021      4      G4  Allegiant Air     SFB   
24413  2021      3      G4  Allegiant Air     SFB   
25965  2021      2      G4  Allegiant Air     SFB   
27504  2021      1      G4  Allegiant Air     SFB   
29079  2020     12      G4  Allegiant Air     SFB   
30610  2020     11      G4  Allegiant Air     SFB   
32301  2020     10      G4  Allegiant Air     SFB   
33621  2020      9      G4  Allegiant Air     SFB   
35198  2020      8      G4  Allegiant Air     SFB   
36617  2020      7      G4  Allegiant Air     SFB   
38049  2020      6      G4  Allegiant Air     SFB   
39428  2020      5      G4  Allegiant Air     SFB   
41021  2020      4      G4  Allegiant Air     SFB   
42742  2020      3      G4  Allegiant Air     SFB   
44523  2020      2      G4  Allegiant Air     SFB   
46301  2020      1      G4  Allegiant Air     SFB   
47595  2019     12      G4  Allegiant Air     SFB   
49897  2019     11      G4  Allegiant Air     SFB   
51656  2019     10      G4  Allegiant Air     SFB   
53786  2019      9      G4  Allegiant Air     SFB   
55171  2019      8      G4  Allegiant Air     SFB   
57474  2019      7      G4  Allegiant Air     SFB   
58697  2019      6      G4  Allegiant Air     SFB   
59900  2019      5      G4  Allegiant Air     SFB   
62151  2019      4      G4  Allegiant Air     SFB   
63884  2019      3      G4  Allegiant Air     SFB   
65593  2019      2      G4  Allegiant Air     SFB   
68098  2019      1      G4  Allegiant Air     SFB   
69050  2018     12      G4  Allegiant Air     SFB   
70770  2018     11      G4  Allegiant Air     SFB   
72480  2018     10      G4  Allegiant Air     SFB   
74170  2018      9      G4  Allegiant Air     SFB   
75752  2018      8      G4  Allegiant Air     SFB   
77287  2018      7      G4  Allegiant Air     SFB   
78507  2018      6      G4  Allegiant Air     SFB   
79715  2018      5      G4  Allegiant Air     SFB   
81020  2018      4      G4  Allegiant Air     SFB   
82268  2018      3      G4  Allegiant Air     SFB   
83448  2018      2      G4  Allegiant Air     SFB   
84755  2018      1      G4  Allegiant Air     SFB   

                                     airport_name  arr_flights  arr_del15  \
689    Sanford, FL: Orlando Sanford International        758.0      226.0   
2395   Sanford, FL: Orlando Sanford International        821.0      313.0   
4105   Sanford, FL: Orlando Sanford International        971.0      412.0   
5794   Sanford, FL: Orlando Sanford International        745.0      248.0   
7478   Sanford, FL: Orlando Sanford International        718.0      224.0   
9178   Sanford, FL: Orlando Sanford International        871.0      258.0   
10882  Sanford, FL: Orlando Sanford International        704.0      250.0   
12602  Sanford, FL: Orlando Sanford International        716.0      240.0   
14362  Sanford, FL: Orlando Sanford International        465.0       94.0   
16102  Sanford, FL: Orlando Sanford International        754.0      273.0   
17820  Sanford, FL: Orlando Sanford International       1165.0      548.0   
19525  Sanford, FL: Orlando Sanford International       1058.0      486.0   
21214  Sanford, FL: Orlando Sanford International        757.0      188.0   
22829  Sanford, FL: Orlando Sanford International        845.0    

In [42]:
sum_OSI_arr_flights = table_of_all_OSI_airport['arr_flights'].sum()
sum_OSI_arr_flights

43073.0

# Total number of delayed flights by Sanford, FL: Orlando Sanford International airport in the last 10 years is 43073.0. Which is significantly lower than the sum of the flights attended by GA: Hartsfield-Jackson Atlanta International airport.

# Inferences

In [49]:
(sum_HJAI_arr_flights/sum_delta_arr_flights)*100

43.292890843047324


### Hence, the sum of the flight delays from the airport with the highest number of delayed flights in a single month is around 43.292% of the sum of the delayed flights from the airlines with the highest number of delayed flights in a single month in the last 10 years.

In [51]:
(sum_AirTran_arr_flights/sum_delta_arr_flights)*100

5.464167919620333

### Therefore the sum of all the delayed flights of the lowest number of delayed flights in a single month is around 5.464% of the sum of the delayed flights from the highest number of flight delay airlines in a single month in the last 10 years.

In [52]:
(sum_OSI_arr_flights/sum_HJAI_arr_flights)*100

1.1527376750362563

### The sum of all the delayed flights of the lowest number of delayed flight airport is around 1.152% of the sum of delayed flights from the highest number of flight delay airports in a single month in the last 10 years